### PR TITLE
[v0.3] [GHA] Migrate v0.3 to GHA

### PR DESCRIPTION
**Pull request by ericpromislow**
Backport PR 383 (migrate v0.5 to GHA)

Related to https://github.com/rancher/rancher/issues/45231

### CODE REVIEW

File: Dockerfile.dapper [hunk 1]
```diff
@@ -3,21 +3,21 @@ FROM registry.suse.com/bci/golang:1.20
...
+ENV HELM_VERSION v3.13.3
 ENV HELM_UNITTEST_VERSION 0.3.2
 
 RUN zypper -n install git docker vim less file curl wget awk
 
 RUN curl -sL https://get.helm.sh/helm-${HELM_VERSION}-linux-${ARCH}.tar.gz | tar xvzf - -C /usr/local/bin --strip-components=1
 
 RUN if [ "${ARCH}" = "amd64" ]; then \
-        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.54.0; \
+        curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.57.1; \
```
Comment (MbolotSuse) [inline Dockerfile.dapper:14]: Why was this upgraded?

**Review by MbolotSuse — COMMENTED** [review]
Questions on a few other changes that were backported that are unrelated to the GHA PR, but outside of that looks good.

File: Dockerfile.dapper (hunk 1 above)
Comment (ericpromislow) [reply Dockerfile.dapper:14]: My usual reason of pulling in newer non-rancher components when backporting

Comment (ericpromislow) [reply Dockerfile.dapper]: And I backed that out.

**Review by nflynt — APPROVED** [review]
This looks good to me. :+1:

**Review by nflynt — APPROVED** [review]

**Review by MbolotSuse — APPROVED** [review]

**Pull request merged**